### DISCUSSION TITLE
Lisäsin itselleni avg+sentit & avg-sentit -toiminnallisuuden

### DISCUSSION THREAD

**Discussion by TMValli** [jisotalo/shelly-porssisahko#62] [General]
Käytän shelly-porssisahköä akkupaketin vuorokausittaisen lataamisen ja purkamisen ohjaukseen ja kaipasin avg-ominaisuuteen laajennusta. Lisäsin avg+ ja avg- toiminnallisuuden. Muutoksessani esim. "avg+1.5" käyttää rajana ("Aina päällä -raja" sekä "Maksimihinta" -kohdissa) päivän keskihinta + 1,5 senttiä tai "avg-1" käyttää rajana päivän keskihintaa yhdellä sentillä vähennettynä.  

En ole aiemmin koskaan ollut missään tekemisisää JavaScript-ohjelmoinnin kanssa ja muutenkin ohjelmointia on viime vuosikymmeninä tullut seurattua ainoastaan sivusta. Homman opiskelu kuitenkin onnistui pitkässä päivässä ChatGPT:n avustuksella ainakin siinä määrin, että sain sen omasta mielestäni toimimaan halutulla tavalla "Jakson halvimmat tunnit" -ohjauksen kanssa.  Muutokset voisi ehkä viedä tarkastettuina viralliseen versioonkin, sillä vastaavaa on kaivannut ilmeisesti näiden keskustelujen perusteella joku muukin.  Uskoakseni lisäys on alaspäin yhteensopiva.

Laitan liitteenä tämänhetkiset muokatut lähdekoodit. Muita havaintoja on ainakin, että avg:n voisi muuttaa kirjainkokoriippumattomaksi ja laittaa hinnan syöttämiset korjaamaan pilkun pisteeksi. Saatan tehdä nämä lisäykset itsekin myöhemmin.

[shelly-porssisahko.zip](https://github.com/user-attachments/files/18233902/shelly-porssisahko.zip)

**Comment by jisotalo** [comment]
Ajatuksena kuulostaa hyvältä, koitanpa keretä tsekkaamaan joku päivä!
